src/browser/provider/index.js

```diff
@@ -289,17 +290,22 @@ export default class BrowserProvider {
         return await this.plugin.maximizeWindow(browserId);
     }
 
-    async takeScreenshot (browserId, screenshotPath, pageWidth, pageHeight) {
+    async takeScreenshot (browserId, screenshotPath, pageWidth, pageHeight, fullPage) {
         const canUseDefaultWindowActions = await this._canUseDefaultWindowActions(browserId);
         const customActionsInfo          = await this.hasCustomActionForBrowser(browserId);
         const hasCustomTakeScreenshot    = customActionsInfo.hasTakeScreenshot;
 
         if (canUseDefaultWindowActions && !hasCustomTakeScreenshot) {
-            await this._takeLocalBrowserScreenshot(browserId, screenshotPath, pageWidth, pageHeight);
-            return;
-        }
+            if (fullPage) {
+                const connection = BrowserConnection.getById(browserId);
 
-        await this.plugin.takeScreenshot(browserId, screenshotPath, pageWidth, pageHeight);
+                connection.addWarning(WARNING_MESSAGE.screenshotsFullPageNotSupported, connection.browserInfo.browserName);
```

Comment:
Thanks to @VasilyStrelyaev, `browserName` contains only the part after the provider name. It's better to use the `alias` property here, it contains the full browser description exactly as it was passed by a user.